osf/management/commands/international_user_metrics.py

```diff
@@ -0,0 +1,200 @@
+# -*- coding: utf-8 -*-
+import logging
+import os
+import csv
+import requests
+import datetime
+import zipfile
+from django.utils import timezone
+from addons.osfstorage.models import UserSettings
+from osf.models import OSFUser
+from django.db.models import F
+from io import BytesIO, StringIO
+
+from website.settings import (
+    USA,
+    GERMANY,
+    CANADA,
+    AUSTRALIA,
+    DS_METRICS_OSF_TOKEN,
+    DS_METRICS_BASE_FOLDER
+)
+
+USER_PER_CSV = 10000
+
+from django.core.management.base import BaseCommand
+
+from framework.celery_tasks import app as celery_app
+
+logger = logging.getLogger(__name__)
+FILENAME = 'osf_international_user_metrics.csv'
+
+
+def get_user_count_by_region(region_name):
+    return UserSettings.objects.filter(default_region__name=region_name).count()
+
+
+def append_to_csv(url, rows, token):
+    headers = {'Authorization': 'Bearer {}'.format(token)}
+    resp = requests.get(url, headers=headers)
+    resp.raise_for_status()
+
+    fieldnames = list(rows[0].keys())
+    file = StringIO(resp.content.decode())
+
+    file.seek(0, os.SEEK_END)
+
+    w = csv.DictWriter(file, fieldnames=fieldnames)
+    w.writerows(rows)
+    file.seek(0)
+    data = file.read()
+
+    resp = requests.put(
+        url,
+        data=data,
+        headers=headers
+    )
+    resp.raise_for_status()
+
+
+def create_csv_header(fieldnames):
+    file = StringIO()
+    writer = csv.DictWriter(file, fieldnames=fieldnames)
+    writer.writeheader()
+    file.seek(0)
+    return file.read()
+
+
+def upload_to_file(url, data, token):
+    headers = {'Authorization': 'Bearer {}'.format(token)}
+    resp = requests.put(
+        url,
+        data=data,
+        headers=headers
+    )
+    resp.raise_for_status()
+
+
+def get_or_create_file(base_folder_url, filename, token, data):
+    headers = {'Authorization': 'Bearer {}'.format(token)}
+
+    resp = requests.get(base_folder_url, headers=headers)
+    resp.raise_for_status()
+
+    base_folder_data = resp.json()['data']
+    children_link = base_folder_data['relationships']['files']['links']['related']['href']
+    resp = requests.get(children_link, headers=headers)
+    resp.raise_for_status()
+
+    children_json = resp.json()['data']
+    try:
+        return next(x['links']['upload'] for x in children_json if x['attributes']['name'] == filename)
+    except StopIteration:
+        resp = requests.put(
+            base_folder_data['links']['upload'] + '?name={}&kind=file'.format(filename),
+            data=data,
+            headers=headers
+        )
+        resp.raise_for_status()
+        return resp.json()['data']['links']['upload']
+
+
+def create_zip(zip_name, file_names, files):
+    buff = BytesIO()
+    with zipfile.ZipFile(buff, mode='w') as zip_file:
+        zip_file.filename = zip_name
```

Comment:
This should write to [a TEMP_FOLDER](https://github.com/CenterForOpenScience/osf.io/pull/9248/files#diff-c39b88b6fef6c8476d857a5c9da960d6R40) in the same way that [the data storage usage script](https://github.com/CenterForOpenScience/osf.io/pull/9248/files#diff-c39b88b6fef6c8476d857a5c9da960d6R560-R561) does, otherwise it will just go into the working directory and will never be cleaned up.

Reply:
Are you sure I was working off [this stackoverflow post](https://stackoverflow.com/questions/34162017/unzip-buffer-with-python) and it didn't seem to create a temp file when I was testing.

Reply:
Oh, so you're saying this is all happening in memory?

Reply:
Yup all buffered via Python3 BytesIO/StringIO magic.